src/js/client.ts

```diff
@@ -59,22 +59,26 @@ export class ReactNativeClient extends BaseClient<ReactNativeClientOptions> {
    * @inheritDoc
    */
   public eventFromMessage(message: string, level?: SeverityLevel, hint?: EventHint): PromiseLike<Event> {
-    return eventFromMessage(this._options.stackParser, message, level, hint, this._options.attachStacktrace).then(
-      (event: Event) => {
-        // TMP! Remove this function once JS SDK uses threads for messages
-        if (!event.exception?.values || event.exception.values.length <= 0) {
+    if (this._options.useThreadsForMessageStack) {
```

Comment:
So by default users will have the new behavior? otherwise, we could invert the if condition

```suggestion
    if (!this._options.useThreadsForMessageStack) {
```